core/src/main/java/org/hibernate/ogm/dialect/eventstate/impl/EventContextManagerInitiator.java

```diff
@@ -30,7 +31,16 @@ private EventContextManagerInitiator() {
 	}
 
 	@Override
-	public EventContextManager initiateService(Map configurationValues, ServiceRegistryImplementor registry) {
-		return new EventContextManager();
+	public EventContextManager initiateService(SessionFactoryImplementor sessionFactory, SessionFactoryOptions sessionFactoryOptions,
+			ServiceRegistryImplementor registry) {
+		EventContextManager eventContextManager = new EventContextManager( registry );
+
+		GridDialect gridDialect = registry.getService( GridDialect.class );
+		// we don't use GridDialects.hasFacet here as we need to set it even for a {@link ForwardingGridDialect}
+		if ( gridDialect instanceof EventContextManagerAwareGridDialect ) {
```

Comment:
Nope, we wan't use the hasFacet machinery as it's not what we want. We also want to inject it in the ForwardingGridDialect if the underlying dialect is not a EventContextManagerAwareGridDialect. See the comment above.